src/components/EmailNotifications/VerifyEmailAddress.js

```diff
@@ -0,0 +1,251 @@
+
+    if (error) {
+      onError()
+    }
+  }, [onError, onReSendVerificationEmail])
+
+  return (
+    <div
+      css={`
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        padding-top: ${3 * GU}px;
+      `}
+    >
+      <div
+        css={`
+          display: flex;
+          flex-direction: column;
+          text-align: center;
+        `}
+      >
+        <img src={emailIllustration} />
+        <span
+          css={`
+            ${textStyle('title2')};
+            margin-top: ${4 * GU}px;
+          `}
+        >
+          Verify your email address
+        </span>
+
+        <span
+          css={`
+            ${textStyle('body2')};
+            color: ${theme.surfaceContentSecondary};
+            margin-top: ${1.5 * GU}px;
+          `}
+        >
+          {`Almost there! We’ve sent a verification email to ${email}.
+          Kindly check your inbox and click the link to verify your account. ${
+            updateMode
+              ? 'Alternatively, you can update this email or delete it, if you wish to unsubscribe'
+              : ''
+          }`}
+        </span>
+        {(() => {
+          if (updateMode) {
+            const buttonWidth = compactMode
+              ? '100% '
+              : `calc((100% - ${2 * GU}px) /  2)`
+            return (
+              <>
+                <div
+                  css={`
+                    margin-top: ${5 * GU}px;
+                  `}
+                >
+                  <Field label="Update email address">
+                    <TextInput
+                      value={emailAddress}
+                      adornment={
+                        emailInvalid === false ? (
+                          <IconCheck
+                            css={`
+                              color: ${theme.positive};
+                            `}
+                          />
+                        ) : emailAddress.trim() ? (
+                          <IconCross
+                            css={`
+                              color: ${theme.negative};
+                            `}
+                          />
+                        ) : (
+                          <IconCheck
+                            css={`
+                              opacity: 0;
+                            `}
+                          />
+                        )
+                      }
+                      adornmentPosition="end"
+                      type="email"
+                      wide
+                      onChange={handleEmailAddressChange}
+                      placeholder="email@address.com"
+                      onBlur={handleEmailAddressBlur}
+                    />
+                    {emailInvalid && (
+                      <p
+                        css={`
+                          color: ${theme.negative};
+                          ${textStyle('body4')};
+                          float: left;
```

Comment:
Is the `float` needed?

Reply:
ahh yes
<img width="586" alt="Screen Shot 2020-04-28 at 8 18 24 PM" src="https://user-images.githubusercontent.com/11763623/80546980-71bfa300-898d-11ea-8ce0-1004e1f9653f.png">

Reply:
Should we use `text-align: left` or flexbox? I try to avoid `float` as it impacts the following elements and the parent.

Reply:
ah yes!  just to understand by impact what do you mean? what kind of error or weird thing generates?

Reply:
having a weird behavior using text-align: left, moves the content above a bit down

Reply:
Would it work with an explicit `height: 0`? By impact I mean that its height doesn’t count to calculate the parent height anymore, and the following elements have a specific layout algorithm being applied to them.